5.0/en/0x12-V3-Session-management.md

```diff
@@ -91,10 +90,88 @@ Attackers start a half-open attack by attempting to lock, reset, or recover a cr
+
+| # | Description | L1 | L2 | L3 |
+| :---: | :--- | :---: | :---: | :---: |
+| **3.8.9** | [ADDED] Verify that Clients are using the response type "code" (aka authorization code grant type) or any other response type that causes the Authorization Server to issue access tokens in the token response, such as the "code id_token" response type. This allows the Authorization Server to detect replay attempts by attackers and generally reduces the attack surface since access tokens are not exposed in the URLs. It also allows the Authorization Server to sender-contrain the issued tokens.  | ✓ | ✓ | ✓ |
+
+
+## Token Replay Prevention 
+
+| # | Description | L1 | L2 | L3 |
+| :---: | :--- | :---: | :---: | :---: |
+| **3.8.10** | [ADDED] Verify that Resource Servers are using mechanisms for sender-constraining access tokens to prevent token replay, such as Mutual TLS for OAuth 2.0 or OAuth Demonstration of Proof of Possession (DPoP).  | ✓ | ✓ | ✓ |
+| **3.8.11** | [ADDED]  Verify that refresh tokens are sender-constrained or use refresh token rotation. | ✓ | ✓ | ✓ |
+
+
+## Access Token Privilege Restriction
+
+| # | Description | L1 | L2 | L3 |
+| :---: | :--- | :---: | :---: | :---: |
+| **3.8.12** | [ADDED] Verify that the privileges associated with an access token should be restricted to the minimum required for the particular application or use case. This prevents clients from exceeding the priviliges authorized by the Resource Owner. It also prevents users from exceeding their privileges authorized by the respective security policy. Privilege restrictions also help to reduce the impact of access token leakage.  | ✓ | ✓ | ✓ |
+| **3.8.13** | [ADDED] Verify that access tokens are restricted to certain Resource Servers (audience restriction), preferably to a single Resource Server. Every Resource Server is obliged to verify, for every request, whether the access token sent with that request was meant to be used for that particular Resource Server. If not, the Resource Server must refuse to serve the respective request. Clients may utilize the parameters "scope" and "resource", respectively to determine the Resource Server they want to access.  | ✓ | ✓ | ✓ |
+| **3.8.14** | [ADDED] Verify that access tokens are restricted to certain resources and actions on Resource Servers or resources. Every Resource Server is obliged to verify, for every request, whether the access token sent with that request was meant to be used for that particular action on the particular resource. If not, the Resource Server must refuse to serve the respective request. Clients may utilize the parameters "scope" and "authorization_details" to determine those resources and/or actions. | ✓ | ✓ | ✓ |
```

Comment:
Do you mean differentiate 3.8.14 from 3.8.13, right? If yes, then I would say 3.8.13 is more focused on the Resource Servers while 3.8.14 is more focused on the actions regarding resources. Moreover, 3.8.13 is requiring Clients to utilize the parameters "scope" and "resource" to determine the Resource Server they want to access while 3.8.14 is requiring Clients to utilize the parameters "scope" and "authorization_details" to determine the desired resources and/or actions. 

I know it seems very similar, and from what I read on the RFC it caught me off guard a few times, but separating them as even highlighted in the RFC seems logical and has some merit to it, in my opinion.

Reply:
Apologies, I was specifically referring to differentiating between 3.8.14 and 3.8.12. Is 3.8.12 just referring to principle of least privilege because if so I think we should talk about that generically in chapter 4 access control but not here.

Reply:
Yes, 3.8.12 specifically deals with principle of least privilege and access controls in relation to the Clients accessing resources. @tghosth, do you want me to add this Oauth requirement in chapter 4 access controls instead of keeping it together as bundled with the rest of the Oauth stuff?

Reply:
I am pretty sure that [4.1.3](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x12-V4-Access-Control.md#v41-general-access-control-design) already covers "principle of least privilege", no? Is there anything else that needs to be added?